src/errors/FailedMiddlewareError.ts

```diff
@@ -0,0 +1,3 @@
+class FailedMiddlewareError extends Error {}
+
```

Comment:
Another design pattern question:

Do we use these blank class definitions in anticipation of fleshing them out in the future? Or are just taking advantage of the fact that by creating new class wrappers around Error we can specify the name without having to write error messages

Reply:
We won't be fleshing it out -- but this lets us define new types of error which we could then handle in the middleware to translate to more distinct HTTP response codes (it also allows us to write tests that ensure that a specific TYPE of error was made).